cirq/devices/grid_qubit.py

```diff
@@ -275,6 +275,26 @@ class GridQubit(_BaseGridQid):
         cirq.GridQubit(5, 4)
     """
 
+    def __init__(self, row: int, col: int):
+        super().__init__(row, col)
+        self._hash = super().__hash__()
+
+    def __hash__(self):
+        # Explicitly cached for performance (vs delegating to Qid).
+        return self._hash
```

Comment:
Suggestion: Move this to `_BaseGridQid` to get some speed benefits for GridQid as well?

Reply:
Can't. The child class may add additional attributes relevant to the hash, so the field can't be initialized in baseGridQid's init method.